subiquitycore/tests/mocks.py

```diff
@@ -23,6 +24,7 @@ class MockedApplication:
     autoinstall_config = {}
     answers = {}
     opts = None
+    set_source_variant = SubiquityServer.set_source_variant
```

Comment:
What's going on here? Even though it's in tests code, I'd rather avoid the dependency from `subiquitycore` to `subiquity`.

Reply:
That's a good point. Originally I had this in place because the identity controller tests needed access to this function through a dependency on the source controller and I thought it better to abstract that logic in mock application than have the identity controller tests need to know what to mock. But that's resolved with #2131 so I'll update this to just patch it in the source controller.

Reply:
I rebased on the updated tests and moved the logic to patch that function in just the source controller tests. Wdyt?

Reply:
Yes, that's much better IMO.